RELEASE.md

```diff
@@ -0,0 +1,19 @@
+# Release Process
+
+The process of creating a release is described in this document. Replace `X.Y.Z` with the version to be released.
+
+## 1. Create a `releases/X.Y.Z` branch 
```

Comment:
I don't see old branches such as `releases/x.y.z`, looks like all releases were created by tags only.

Reply:
right, do we want to change that so the process will be aligned with the other projects?